### PR TITLE
add beaker-py

### PR DESCRIPTION
**Pull request by h-vetinari**
Needed for [tango](https://github.com/conda-forge/tango-feedstock/pull/12).

The name reflects the upstream name for the python bindings, though we couldn't package the underlying `beaker` (written in Go apparently) as-is because there's https://github.com/conda-forge/beaker-feedstock already.

### CODE REVIEW

**Comment by h-vetinari** [comment]
@dirkgr @epwalsh

Could you comment on the failing test suite here? AFAICT it makes little sense to try to run docker-based tests here (not least since conda-forge's [docker-py](https://github.com/conda-forge/docker-py-feedstock) actually doesn't install a docker-daemon).

**Comment by epwalsh** [comment]
The test failures are because many of the tests require a Beaker personal access token. I hate to say it but the best option might be to just not run any of the unit tests.

**Comment by h-vetinari** [comment]
> The test failures are because many of the tests require a Beaker personal access token. I hate to say it but the best option might be to just not run any of the unit tests.

Would it not make sense (medium-term & upstream) to make those tests conditional on the presence of the token, and skip them if it's not present?

**Comment by epwalsh** [comment]
> Would it not make sense (medium-term & upstream) to make those tests conditional on the presence of the token, and skip them if it's not present?

That's a possibility, but I understated how many tests require a token... it's pretty much 100% of the tests. Though I've proposed an idea internally (https://github.com/allenai/beaker/issues/2239 - repo is private so you won't be able to view the issue, but I'm linking here for my own records) that would allow the Beaker server to issue temporary tokens that could be used by GitHub runners via approved repositories. If that gets implemented then we wouldn't have to skip any tests.

**Comment by h-vetinari** [comment]
> If that gets implemented then we wouldn't have to skip any tests.

That sounds great, thanks! I like not skipping tests. :)

**Comment by h-vetinari** [comment]
@conda-forge/help-python

This should be ready for review. :)

File: recipes/beaker-py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+source:
+  url: https://github.com/allenai/beaker-py/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: d1445637651e9a6ddbcce609108284a4d6b4ad3603e579a96dbdc1e77d9e0447
+
+build:
+  skip: true  # [win]
+  number: 0
+  script:
+    - rm pyproject.toml
```
Comment (jaimergp) [inline recipes/beaker-py/meta.yaml:15]: Why is this needed? It only lists the build backend for `pip install`.

```suggestion
```

File: recipes/beaker-py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+package:
+  name: beaker-py
+  version: {{ version }}
+
+source:
+  url: https://github.com/allenai/beaker-py/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: d1445637651e9a6ddbcce609108284a4d6b4ad3603e579a96dbdc1e77d9e0447
+
+build:
+  skip: true  # [win]
```
Comment (jaimergp) [inline recipes/beaker-py/meta.yaml:12]: Doesn't this package qualify for `noarch: python`?

```suggestion
  noarch: python
```

**Review by jaimergp — CHANGES_REQUESTED** [review]
Just a couple questions!

**Comment by h-vetinari** [comment]
Thanks for the review @jaimergp, this should be ready now I hope.

**Pull request merged**